File: COMPILER DESIGN/compiler_design_04.py

road = dict()
state = input().split(' ')
parameter = input().split(' ')

for i in state:
    temp = dict()
    for j in parameter:
        print(f'{i} {j}:', end=' ')
        value = input()
        temp[j] = value
    road[i] = temp

roadmap = [state[0]]
str = input()

for i in str:
    try:
        current = roadmap[-1]
        next = road[current][i]
        roadmap.append(next)
    except:
        print("not valid")

if roadmap[-1] == state[-1]:
    print(roadmap)
else:
    print('reject')


# road = {
#     0: {'a': 1, 'b': 0},
#     1: {'a': 1, 'b': 2},
#     2: {'a': 1, 'b': 3},
#     3: {'a': 1, 'b': 0}
# }

# 0 1 2 3
# a b
# 0 a: 1
# 0 b: 0
# 1 a: 1
# 1 b: 2
# 2 a: 1
# 2 b: 3
# 3 a: 1
# 3 b: 0
# aabb

File: COMPILER DESIGN/compiler_design_04.cpp
#include <bits/stdc++.h>
using namespace std;


int main()
{
    int road[4][2];
    road[0][0] = 1;
    road[0][1] = 0;
    road[1][0] = 1;
    road[1][1] = 2;
    road[2][0] = 1;
    road[2][1] = 3;
    road[3][0] = 1;
    road[3][1] = 0;

    string roadmap = "0";
    string str;
    cin >> str;

    for(int i = 0; i < str.length(); i++) {
        int current = (int) roadmap[i] - (int) '0';
        int idx = (int) str[i] - (int) 'a';
        int next = road[current][idx];
        roadmap += to_string(next);
    }

    if(roadmap.back() == '3')
        cout << roadmap << endl;
    else
        cout << "reject" << endl;

    return 0;
}